### DISCUSSION TITLE
测试使用IPv6 UDP Client

### DISCUSSION THREAD

**Discussion by leiBravo** [cloudwu/skynet#1765] [General]
我需要使用使用skynet ipv6 udp client的功能，我修改了test/testudp.lua，代码如下
```
local skynet = require "skynet"
local socket = require "skynet.socket"

local function client()
	local c = socket.udp(function(str, from)
		print("client recv", str, socket.udp_address(from))
	end)
	socket.udp_connect(c, "fd53:7cb8:383:3::13a", 13400)
	socket.write(c, "hello")
end

skynet.start(function()
	skynet.fork(client)
end)
```
环境为Ubuntu20.04 x86_64，本地的IPv6地址为fd53:7cb8:383:3::10，能够ping通服务端fd53:7cb8:383:3::13a。
我更新了最新的代码，编译好了skynet，在第一个终端执行./skynet examples/config，在第二个终端执行telnet 127.0.0.1 8000，进入了skynet的控制台。然后执行了修改后的测试代码 start testudp。在第一个终端中报下面的错误
```
[:00000000] socket-server: udp socket (9) type mismatch.
[:01000010] socket: accpet error: protocol mismatch
```
原因是udp.socket在创建socket的时候没有指定IPv6，默认是IPv4，而connect的时候用的却是IPv6的地址。我应该怎么修改testudp.lua，让它能连上IPv6的服务端？

**Comment by cloudwu** [comment]
你需要在 socket.udp() 的第二三个参数绑定正确的 ipv6 地址。
https://github.com/cloudwu/skynet/blob/master/lualib/skynet/socket.lua#L455
